sdk/src/lib/api/HttpApi.ts

```diff
@@ -1,11 +1,48 @@
+import { backoffRetrier, RetryOptions } from "../utils"
+
 /**
  * Represents an abstract HTTP API.
  */
 export default abstract class HttpApi {
   #apiUrl: string
 
-  constructor(apiUrl: string) {
+  /**
+   * Retry options for API requests.
+   */
+  #retryOptions: RetryOptions
+
+  constructor(
+    apiUrl: string,
+    retryOptions: RetryOptions = {
+      retries: 5,
+      backoffStepMs: 1000,
+    },
+  ) {
     this.#apiUrl = apiUrl
+    this.#retryOptions = retryOptions
+  }
+
+  /**
+   * Makes an HTTP request with retry logic.
+   * @param requestFn Function that returns a Promise of the HTTP response.
+   * @returns The HTTP response.
+   * @throws Error if the request fails after all retries.
+   */
+  async requestWithRetry(
```

Comment:
What do you think about making this fn private and using it in `getRequest` and `postRequest`?

Reply:
I thought about it for a while. It may be worth to add retries by default.